### DISCUSSION TITLE
How to insert multiple rows using the serverless driver?

### DISCUSSION THREAD

**Discussion by PhilBookst** [planetscale/discussion#382] [Q&A] [answered]
Hey, 

how can I insert multiple rows using the serverless driver?

I've tried queries like these but couldn't get it to work:
```ts
    const result = await conn.execute(
      "INSERT INTO Table (col1, col2) VALUES (?, ?)",
      [
        ["val1a","val2a"],
        ["val2a", "val2b"],
      ]
    );
```

Any help is much appreciated! This seems like a very trivial feature

**Comment by nickvanw** [comment, marked as answer]
Hi @PhilBookst! This is possible today, you'll just have to change how you're writing the query and your arguments slightly:

```javascript
    const result = await conn.execute(
      "INSERT INTO Table (col1, col2) VALUES (?, ?), (?,?)",
      ["val1a","val2a","val2a", "val2b"]
    );
```

**Reply by PhilBookst** [reply]
Ah ok, so say I want to insert 500 rows at once I'd have to do it like this? 
```ts
const query = "INSERT INTO Table (col1, col2) VALUES " + Array(500).fill("?,?").join(", ")
const values = [...data]
const result = await conn.execute(query, values)
```

**Reply by nickvanw** [reply]
@PhilBookst that's exactly it! 

The `database.js` driver is not doing much more than passing through the query and data arguments, so you'll need to construct a valid query with the same number of `?`s as values.
